kitodo-mediaserver-core/src/main/java/org/kitodo/mediaserver/core/util/FileDeleter.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+package org.kitodo.mediaserver.core.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileTime;
+import java.time.Instant;
+import java.util.stream.Stream;
+import org.springframework.util.FileSystemUtils;
+
+/**
+ * Deletes files (and folders).
+ */
+public class FileDeleter {
+
+    /**
+     * Deletes everything in a path.
+     * @param path path to a directory which should be cleared
+     * @throws IOException on access errors
+     */
+    public void delete(Path path) throws IOException {
+        FileSystemUtils.deleteRecursively(path);
+    }
+
+    /**
+     * Deletes all files in a path (and empty folders) that are older than given time.
+     * @param path path to a directory which should be cleared
+     * @param age timespam since last modified in seconds
+     * @throws IOException on access errors
+     */
+    public void delete(Path path, Long age) throws IOException {
+
+        // get items in the directory
+        Stream<Path> items = Files.list(path);
+
+        Path item;
+        while (items.iterator().hasNext()) {
+            item = items.iterator().next();
+            if (Files.isDirectory(item)) {
+                // recursively delete directory child items
+                delete(item, age);
+            } else {
+                // check the modified date and delete if older than limit
+                Instant limit = Instant.now().minusSeconds(age);
```

Comment:
You could instantiate the limit ouside the loop instead of doing it for each file